File: Enigma-Machine-master/src/me/matt/enigma/wrappers/PlugBoardSwitch.java
package me.matt.enigma.wrappers;

public class PlugBoardSwitch {

    char in;
    char out;

    public PlugBoardSwitch(final char in, final char out) {
        this.in = in;
        this.out = out;
    }

    public boolean checkSwap(final char c) {
        return c == in || c == out;
    }

    public char swap(final char c) {
        if (c == in) {
            return out;
        }
        return c == out ? in : c;
    }

}

File: app/src/main/java/com/sdr/enigmasimulator/enigma/PlugBoardSwitch.kt
package com.sdr.enigmasimulator.enigma

class PlugBoardSwitch(var `in`: Char, var out: Char) {
    fun checkSwap(c: Char): Boolean {
        return c == `in` || c == out
    }

    fun swap(c: Char): Char {
        if (c == `in`) {
            return out
        }
        return if (c == out) `in` else c
    }

}
